File: src/main/java/com/apnaghar/service/NotificationService.java
package com.apnaghar.service;

import com.apnaghar.model.Notification;

import java.util.List;

public interface NotificationService {

    // Create notification
    void createNotification(Long userId, String role, String message);

    // Get notifications for logged-in user
    List<Notification> getMyNotifications(String email);

    // Mark notification as read
    void markAsRead(Long notificationId, String email);

    // Unread count
    long getUnreadCount(String email);
    
    //for mark as all read notification
    void markAllAsRead(String email);

}

File: apnaghar-frontend/src/services/notificationService.js
import api from "./api";

export const getMyNotifications = () => {
  return api.get("/notifications/my");
};

export const markNotificationRead = (id) => {
  return api.put(`/notifications/${id}/read`);
};

export const getUnreadCount = () => {
  return api.get("/notifications/unread-count");
};

export const markAllRead = () => {
  return api.put("/notifications/read-all");
};
